cmd/kube-proxy/app/server.go

```diff
@@ -136,10 +138,19 @@ func NewProxyServerDefault(config *options.ProxyServerConfig) (*ProxyServer, err
 		protocol = utiliptables.ProtocolIpv6
 	}
 
+	var netshInterface utilnetsh.Interface
+	var iptInterface utiliptables.Interface
+	var dbus utildbus.Interface
+
 	// Create a iptables utils.
 	execer := exec.New()
-	dbus := utildbus.New()
-	iptInterface := utiliptables.New(execer, dbus, protocol)
+
+	if runtime.GOOS == "windows" {
```

Comment:
I dislike all these checks.  Can we find a way to do this more abstractly.  Maybe something like `platformParams := createPlatformParams()` and make that exist in distinct _windows and _linux files?